platform/platform.go

```diff
@@ -1,31 +1,131 @@
-// +build linux darwin
+//go:build !android
+// +build !android
 
 package platform
 
 import (
 	"fmt"
 	"os/exec"
 	"regexp"
+	"runtime"
 	"strings"
+
+	"github.com/DataDog/gohai/utils"
 )
 
-// GetArchInfo() returns basic host architecture information
-func GetArchInfo() (archInfo map[string]interface{}, err error) {
-	archInfo = make(map[string]interface{})
+// Platform holds metadata about the host
+type Platform struct {
+	// GoVersion is the golang version.
+	GoVersion string
+	// PythonVersion is the version of python in the current env (ie: returned by "python -V").
+	PythonVersion string
+	// GoOS is equal to "runtime.GOOS"
+	GoOS string
+	// GoArch is equal to "runtime.GOARCH"
+	GoArch string
+
+	// KernelName is the kernel name (ex:  "windows", "Linux", ...)
+	KernelName string
+	// KernelRelease the kernel release (ex: "10.0.20348", "4.15.0-1080-gcp", ...)
+	KernelRelease string
+	// Hostname is the hostname for the host
+	Hostname string
+	// Machine the architecture for the host (is: x86_64 vs arm).
+	Machine string
```

Comment:
Hmm... why not just use `Arch`/`Architecture` for this field?

Reply:
Since only new functionality will use the new structs, there's no real need to retain old naming other than historic consistency. It's up to you what you want to do here but if we can fix up the field names moving forward, especially for named ones added now, I would be a strong proponent of the change otherwise we will be stuck with odd naming for even longer. WDYT?

Reply:
Knowing that we're trying to reproduce the content  of the metadata payload who embed gohai I think keeping the name is a good idea. Once we move gohai to the agent code base we can rename everything we want.